src/common/transformations/src/transformations/utils/utils.cpp

```diff
@@ -133,6 +138,31 @@ bool has_f16_constants(const std::shared_ptr<const ov::Model>& function) {
     return false;
 }
 
+bool is_large_language_model(const std::shared_ptr<const ov::Model>& model) {
+    using namespace ov::pass::pattern;
+
+    const auto past = wrap_type<ov::op::v6::ReadValue>();
+    const auto convert_past = wrap_type<ov::op::v0::Convert>({past});
+    const auto gather_input = std::make_shared<ov::pass::pattern::op::Or>(OutputVector{past, convert_past});
+    const auto beam_idx = wrap_type<ov::op::v0::Parameter>();
+    const auto gather_past =
+        wrap_type<ov::op::v8::Gather>({gather_input, beam_idx, wrap_type<ov::op::v0::Constant>()});
+    const auto gather_convert = wrap_type<ov::op::v0::Convert>({gather_past});
+    const auto concat_past_input =
```

Comment:
can we use optional here as well?

Reply:
The current logic supports the following inputs of `concat`:
-`ReadValue`
-`ReadValue -> Convert`
-`ReadValue -> Gather`
-`ReadValue -> Convert -> Gather`
-`ReadValue -> Gather -> Convert`
-`ReadValue -> Convert -> Gather -> Convert`

To avoid situations with matching on `ReadValue -> Convert -> Convert`, I left `Gather` as `wrap_type` and `Or` before `Concat`.

Thanks!